src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,844 @@
+        updated_providers = current_providers + [provider_location]
+
+        # Write the updated DefaultOTAProviders list back to the Requestor
+        attr = self.cluster_otar.Attributes.DefaultOTAProviders(value=updated_providers)
+        resp = await controller.WriteAttribute(
+            attributes=[(0, attr)],
+            nodeid=requestor_node_id
+        )
+        logger.info(f'Prerequisite #4.0 - Write DefaultOTAProviders response: {resp}')
+        asserts.assert_equal(resp[0].Status, Status.Success, "Failed to write DefaultOTAProviders attribute")
+
+    async def is_port_free(self, port):
+        """Check if a TCP port is free"""
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            return s.connect_ex(('127.0.0.1', port)) != 0
+
+    async def write_acl(self, controller, node_id, acl):
+        """
+        Writes the Access Control List (ACL) to the DUT device using the specified controller.
+
+        Args:
+            controller: The Matter controller (e.g., th1, th4) that will perform the write operation.
+            acl (list): List of AccessControlEntryStruct objects defining the ACL permissions to write.
+            node_id:
+
+        Raises:
+            AssertionError: If writing the ACL attribute fails (status is not Status.Success).
+        """
+        result = await controller.WriteAttribute(
+            node_id,
+            [(0, Clusters.AccessControl.Attributes.Acl(acl))]
+        )
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+        return True
+
+    async def run_ota_step(
+        self,
+        step_number,
+        controller,
+        requestor_node_id,
+        requestor_discriminator,
+        requestor_passcode,
+        requestor_port,
+        provider_node_id,
+        provider_discriminator,
+        provider_setupPinCode,
+        provider_port,
+        provider_ota_file,
+        fabric_id,
+        provider_queue="Busy",   # Optional
+        provider_timeout=120         # Optional
+    ):
+        """
+        Run a full OTA step: launch Requestor, Provider, Commissioning, ACLs, AnnounceOTAProvider.
+        """
+
+        # ------------------------------------------------------------------------------------
+        # Prerequisites: All the steps to prepare the Requestor and Provider for OTA update
+        # ------------------------------------------------------------------------------------
+
+        # Prerequisite #1.1 - Launch Requestor (DUT) from the test
+        logger.info(f'{step_number}: Prerequisite #1.1 - Launching Requestor (DUT) from test')
+        requestor_proc = await self.launch_ota_requestor(
+            discriminator=requestor_discriminator,
+            passcode=requestor_passcode,
+            secured_device_port=requestor_port
+        )
+        await asyncio.sleep(5)
+
+        # Prerequisite #1.2 - Commission the Requestor using controller
```

Comment:
So the requestor here is the DUT and it will be commissioned at the start of the test. You shouldn't need to recommission it in the middle of the test unless you're asking the tester to factory reset the device, which this test does not do.